FinancialTransactionsApi/V1/Gateways/PostgreDbGateway.cs

```diff
@@ -38,7 +38,31 @@ public async Task<Transaction> GetTransactionByIdAsync(Guid id)
             return await Task.FromResult(response.FirstOrDefault()?.ToDomain()).ConfigureAwait(false);
         }
 
-        public Task<PagedResult<Transaction>> GetPagedTransactionsAsync(TransactionQuery query) => throw new NotImplementedException();
+        public async Task<Paginated<Transaction>> GetPagedTransactionsAsync(TransactionQuery query)
+        {
+            var spec = new GetTransactionByDateSpecification(query.StartDate ??= new DateTime(), query.EndDate ??= DateTime.Now);
+
+            var response = _databaseContext.Transactions.Where(spec.Criteria);
+
+            if (query.TransactionType.HasValue)
+            {
+                response = response.Where(x => x.TransactionType == query.TransactionType.ToString());
+            }
+
+            var page = query.Page > 1 ? query.Page : 0;
+
+            var filteredResponse = response.Skip(page * query.PageSize).Take(query.PageSize).Select(x => x.ToDomain());
+
+            var result = await Task.FromResult(filteredResponse.AsEnumerable()).ConfigureAwait(false);
+
+            return new Paginated<Transaction>()
```

Comment:
Ideally the responsibility of the Gateway class is to call the database logic for querying and return the results to the UseCase class.
The responsibility of the UseCase class would be to do all the mapping to Paginated Results and returning that to the Controller.

Reply:
I followed the logic of the GetAllActive method. Would you suggest the method just returns the `IEnumerable<Transactions>` and then the other logic is done in the UserCase?

Reply:
yes please